portfolio-project-matching-app/backend/dao.js

```diff
@@ -249,42 +249,40 @@ const createAssociationInputIsValid = async (coll, id1, coll1, id2, coll2) => {
         getDocSnapshotById(coll, `${id1}_${id2}`)
     ]);
     // handle case where coll does not exist in database
-    if (collSnap.empty) {
+    if (!associationCollIsValid(coll)) {
         console.log(`Collection '${coll}' does not exist`);
         return false;
     }
     // handle case where id1 does not exist in coll1
-    else if (!id1Snap.exists()) {
+    if (!id1Snap.exists()) {
         console.log(`Invalid '${coll1}' id: '${id1}' does not exist`);
         return false;
     }
     // handle case of projects_users where project is not open
-    else if (coll === 'projects_users' && !id1Snap.data().open) {
+    if (coll === 'projects_users' && !id1Snap.data().open) {
         console.log(`Invalid '${coll1}' id: ${id1} is at capacity`);
         return false;
     }
     // handle case of applications where user is already added to project
-    else if (coll === 'applications') {
-        const projectsUsersSnap = getDocSnapshotById('projects_users', `${id1}_${id2}`);
+    if (coll === 'applications') {
+        const projectsUsersSnap = await getDocSnapshotById('projects_users', `${id1}_${id2}`);
```

Comment:
Originally, when I was using `if`, `else if`, `else`: I failed to return true in the case of `coll === 'applications'`. Swapped out for series of ifs because they all manifest in a return (so not other branch would need be hit).